### PR TITLE
[READY] Fasplosions II: Attack of the Maxcaps

### PR DESCRIPTION
**Pull request by cerebral-314**
## About The Pull Request

This will make explosions almost instant.
Credits: https://github.com/tgstation/tgstation/pull/50594

## Why It's Good For The Game

1.  You won't be able to outrun maxcaps anymore.
2.  People will actually die when the nuke goes off.
3.  It'll reduce the lag on the server caused by explosions.
4.  It's the best thing since deep-fried maxcaps.

## Changelog
:cl:
refactor: turned explosions datum into it's own subsystem.
tweak: explosions are now instant(no more running away from maxcaps).
add: added a queuing system for rebuilding pipelines to balance overhead.
/:cl:

### CODE REVIEW

**Comment by Flakkyboo** [comment]
Kinda liked outrunning explosions especially if I heard the beep before it went off

**Comment by utilbird** [comment]
I will miss the nuke wall of death

File: code/modules/events/spacevine.dm [hunk 1]
```diff
@@ -146,7 +146,13 @@
 	quality = NEGATIVE
 
 /datum/spacevine_mutation/aggressive_spread/on_spread(obj/structure/spacevine/holder, turf/target)
-	target.ex_act(severity, null, src) // vine immunity handled at /mob/ex_act
```
Comment (Angustmeta) [inline code/modules/events/spacevine.dm:149]: This change makes aggressive spreading space vines explode turfs and deal friendly fire/kill vine immune mobs. I later fixed that on TG with this PR:
https://github.com/tgstation/tgstation/pull/52650

Comment (cerebral-314) [reply code/modules/events/spacevine.dm:149]: Fixed thank you for the heads up.

File: code/modules/events/spacevine.dm [hunk 2]
```diff
@@ -112,6 +112,9 @@
 		. = 1
 		QDEL_IN(holder, 5)
 
+/datum/spacevine_mutation/aggressive_spread/proc/aggrospread_act(obj/structure/spacevine/S, mob/living/M)
+	return
+
```
Comment (Angustmeta) [inline code/modules/events/spacevine.dm:117]: Now aggressive spreading won't work at all. This PR needs to leave the spacevine.dm file alone and keep ex_act as placeholder.

**Comment by LordSaladin** [comment]
Are we trying to dethrone unitystation by having rounds end faster

**Comment by Isy232** [comment]
Holy shit merge now!

**Comment by yorii** [comment]
> Are we trying to dethrone unitystation by having rounds end faster

Why would non-laggy explosions make the round end faster?

**Comment by Isy232** [comment]
> > Are we trying to dethrone unitystation by having rounds end faster
> 
> Why would non-laggy explosions make the round end faster?

Due to explosion lag you can often run out of the radius of an explosion before it hits you.

I've seen maxcaps go off and the guy 2 tiles away runs out of the explosion.

I still think this is a good idea though, its bullshit for explosive weapons to be either useful or shit depending on server lag.

**Review by qwertyquerty — DISMISSED** [review]
Looked through it, probably good for a TM on golden

File: code/datums/martial/wrestling.dm [hunk 3]
```diff
@@ -417,7 +417,7 @@
 
 		if (falling == 1)
 			if (prob(33) || D.stat)
-				D.ex_act(EXPLODE_LIGHT)
+				SSexplosions.lowturf += D
```
Comment (PowerfulBacon) [inline code/datums/martial/wrestling.dm]: For all of these being replaced with SSexplosions.lowturf += D, I feel like keeping the ex_act proc would be better an have the proc call SSexplosions.lowobj += src depending on what you call it from, and then obviously override it on turfs to add itself to low/med/highturf depending on the severity passed in. Would make porting anything with explosions easier

Comment (cerebral-314) [reply code/datums/martial/wrestling.dm]: I can do that... just figured accessing a list would be less intensive then calling an entire proc that just does the same thing.

**Comment by PowerfulBacon** [comment]
Merge conflicts

**Comment by RafRoq** [comment]
Noooooooooo why can't I abuse time dilation to run from devastation range anymore

**Comment by PowerfulBacon** [comment]
Large explosions like the nuke freeze the server for like 4-5 seconds, but it works fine for small explosions

**Comment by eLeCtrOssSnake** [comment]
Please no...

**Pull request closed without merging**